libs/core/graphql/src/apollo/queued-apollo.service.spec.ts

```diff
@@ -100,25 +100,50 @@ describe('Core | GraphQL | DaffQueuedApollo', () => {
     })
 
     describe('when multiple requests are made', () => {
-      it('should make the second request after the first one completes', () => {
-        testScheduler.run(({cold, expectObservable}) => {
-          apollo.mutate.withArgs(jasmine.objectContaining({
-            mutation: req0
-          })).and.returnValue(cold('--a', {a: data0}))
-          apollo.mutate.withArgs(jasmine.objectContaining({
-            mutation: req1
-          })).and.returnValue(cold('--a', {a: data1}))
-
-          ob0 = service.mutate({
-            mutation: req0
-          });
-          ob1 = service.mutate({
-            mutation: req1
-          });
+      describe('and the first one completes successfully', () => {
+        it('should make the second request after the first one completes', () => {
+          testScheduler.run(({cold, expectObservable}) => {
+            apollo.mutate.withArgs(jasmine.objectContaining({
+              mutation: req0
+            })).and.returnValue(cold('--a', {a: data0}))
+            apollo.mutate.withArgs(jasmine.objectContaining({
+              mutation: req1
+            })).and.returnValue(cold('--a', {a: data1}))
+
+            ob0 = service.mutate({
+              mutation: req0
+            });
+            ob1 = service.mutate({
+              mutation: req1
+            });
+
+            expectObservable(ob0).toBe('--(a|)', {a: data0});
+            expectObservable(ob1).toBe('----(a|)', {a: data1});
+          })
+        });
+      });
 
-          expectObservable(ob0).toBe('--(a|)', {a: data0});
-          expectObservable(ob1).toBe('----(a|)', {a: data1});
-        })
+      describe('and the first one throws an error', () => {
+        it('should make the second request after the first one throws an error', () => {
+          testScheduler.run(({cold, expectObservable}) => {
+            apollo.mutate.withArgs(jasmine.objectContaining({
+              mutation: req0
+            })).and.returnValue(cold('--#', {}, 'error'))
+            apollo.mutate.withArgs(jasmine.objectContaining({
+              mutation: req1
+            })).and.returnValue(cold('--a', {a: data1}))
+
+            ob0 = service.mutate({
+              mutation: req0
+            });
+            ob1 = service.mutate({
+              mutation: req1
+            });
+
+            expectObservable(ob0).toBe('--#', {}, 'error');
```

Comment:
@damienwebdev @lderrickable  Should the outside observable, the one returned by `QueuedApollo#mutate`, complete on error (`'--(#|)'` or is it fine like this?

Reply:
I suspect that all observables from Apollo complete upon an error, so I think it should be safe to complete.

Reply:
I'm not sure that rxjs even supports completing after error like this but I called `complete` anyway.

Reply:
http://reactivex.io/documentation/contract.html

```
OnError
indicates that the Observable has terminated with a specified error condition and that it will be emitting no further items
```

Reply:
It should actually be sufficient as you previously had it.

Reply:
I just reverted it to how I previously had it. The test cases are the same anyway.